src/julia-syntax.scm

```diff
@@ -3071,12 +3089,17 @@ f(x) = yt(x)
             `(block (= ,temp ,(renumber-assigned-ssavalues t)) ,ex)
             ex))))
 
+(define (capt-var-access var fname opaq)
+  (if opaq
+      `(call (core getfield) (call (core getfield) ,fname (inert env)) ,(get opaq var))
```

Comment:
getfield of the first arg is treated specially in the closure invocation, so the extra getfield env isn't actually needed

Reply:
Where does that happen? It seems to be working with this.

Reply:
```
julia> let x=1
           @opaque ()->x
       end
()::Any->◌

julia> ans()
ERROR: type Tuple has no field env
Stacktrace:
 [1] (::Core.OpaqueClosure{Tuple{}, Any})()
   @ Base ./opaque_closure.jl:3
 [2] top-level scope
   @ REPL[4]:1
```

What example are you trying? Maybe my support for this is incomplete

Reply:
```
julia> f(x) = @opaque y->x+y
f (generic function with 1 method)

julia> f(2)(10)
12
```

Reply:
Though I also get
```
julia> f(x) = @opaque ()->x
f (generic function with 1 method)

julia> f(2)()
Internal error: encountered unexpected error in runtime:
MethodError(f=typeof(Core.Compiler.push!)(), args=(Core.Compiler.BitSet(bits=Array{UInt64, (0,)}[], offset=-1152921504606846976), :(:env)), world=0x0000000000001064)
jl_method_error_bare at /home/jeff/src/julia/src/gf.c:1766
jl_method_error at /home/jeff/src/julia/src/gf.c:1784
jl_lookup_generic_ at /home/jeff/src/julia/src/gf.c:2354 [inlined]
jl_apply_generic at /home/jeff/src/julia/src/gf.c:2369
analyze_env_uses at ./compiler/ssair/passes.jl:1205
opaque_closure_optim_pass! at ./compiler/ssair/passes.jl:1220
```

Reply:
Thanks, I'll take a look and try to fix this in the morning.